### PR TITLE
doc enhancements

### PR DESCRIPTION
**Pull request by cbielow**
- lots of fixes affecting rendering
- fixes for correctness (referencing the correct lines etc)
- more elaborate explanation of terms in texts (protein accession etc)
- sort glossary for easier maintenance
- update dev build instructions for pyOpenMS
- glossary fixes: improved descriptions
- improve QuantData tutorial

### CODE REVIEW

File: docs/source/user_guide/oligonucleotides_rna.rst [hunk 1]
```diff
@@ -140,14 +152,29 @@ sequence as follows:
...
+    False
 
 DNA, RNA and Protein
 ********************
 
 We can also work with DNA and RNA sequences in combination with the BioPython
 library (you can install BioPython with ``pip install biopython``):
 
-.. code-block:: pseudocode
+.. code-block:: python
```
Comment (jpfeuffer) [inline docs/source/user_guide/oligonucleotides_rna.rst:177]: You probably have to be careful. All python code-blocks will be tested. Not sure if this code is functional.
Just saying, you might need to fix it in CI.

Comment (cbielow) [reply docs/source/user_guide/oligonucleotides_rna.rst]: I reverted to pseudocode, just to avoid installing biopython, which is rather large

Comment (jpfeuffer) [reply docs/source/user_guide/oligonucleotides_rna.rst]: yep. fine with me

**Comment by cbielow** [comment]
building the snippets fails in areas I have not touched.
e.g. https://github.com/OpenMS/pyopenms-docs/actions/runs/12162622521/job/33920001766?pr=449#step:8:84
This probably means that pyopenms-docs has not been build for quite some time. 
It seems pyopenms has changed and now breaks the snippets?

**Comment by jpfeuffer** [comment]
Yes, or they were never touched since CI was introduced.
I think we always just had incremental CI.
No one ever tested everything.

**Comment by cbielow** [comment]
tests pass.
will merge

**Pull request merged**